hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleSystemContractOperations.java

```diff
@@ -94,11 +101,33 @@ public ContractCallRecordBuilder externalizePreemptedDispatch(
     public void externalizeResult(
             @NonNull final ContractFunctionResult result, @NonNull final ResponseCodeEnum responseStatus) {
         final var childRecordBuilder = context.addChildRecordBuilder(ContractCallRecordBuilder.class);
-        childRecordBuilder
-                .transaction(Transaction.DEFAULT)
-                .contractID(result.contractID())
-                .status(responseStatus)
-                .contractCallResult(result);
+
+        final var body = TransactionBody.newBuilder()
```

Comment:
Did we consider dispatching a `UtilPrngTransactionBody` instead of using this one-off pattern?

Reply:
Okay, now this logic is only for the failed scenario. I'm using this pattern here instead of using dispatch because as far as I can see it can't work with the dispatch. If I dispatch a UTIL_PRGN it will always generate prng number but in the case when it's failing we don't want that. We only want to have an empty field.

The other option I can think of is If I dispatch `UtilPrngTransactionBody` with some negative `range` and then change the `UtilPrngHandler` to check if the range is negative then just create an empty prng number field or something like that. But this doesn't sit well with me 🤷 I'm open to discussions here.

Reply:
Ah I see @vtronkov...what about using `systemContractOperations().externalizePreemptedDispatch()` as [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersCall.java#L118)?

Reply:
Good point, I've missed that method. Fixed 👍